lib/src/local_working_copy.rs

```diff
@@ -1204,28 +1201,23 @@ impl TreeState {
         Ok(FileState::for_file(executable, size, &metadata))
     }
 
-    #[cfg_attr(windows, allow(unused_variables))]
     fn write_symlink(&self, disk_path: &Path, target: String) -> Result<FileState, CheckoutError> {
-        #[cfg(windows)]
-        {
-            self.write_file(disk_path, &mut target.as_bytes(), false)
-        }
-        #[cfg(unix)]
-        {
-            let target = PathBuf::from(&target);
-            symlink(&target, disk_path).map_err(|err| CheckoutError::Other {
+        let target = PathBuf::from(&target);
+        try_symlink(&target, disk_path).map_err(|inner_err| match inner_err {
+            SymlinkError::DeveloperModeNotEnabled => CheckoutError::SymlinkNotSupported(inner_err),
```

Comment:
This shouldn't happen since we checked the registry, right? I want to avoid errors when updating the working copy whether or not developer mode is enabled. I think it's fine if this only happens if the user had enabled developer mode, then ran `jj new <something>` and disabled developer mode while that was running. But that's such a rare case that I don't think we need a special message for it; just map it to an io::Error.

Reply:
Originally this error variant existed for tests to use, but the tests can now use check_symlink_support() instead. I've removed it.